### PR TITLE
Refactor editor to use dispatcher for Note store manipulation

### PR DESCRIPTION
**Pull request by d4ncer**
Changes:
- Make Notes and Notebooks available throughout the app as props rather than state variables
- Move createBlock & updateBlock functionality out of React component and into Backbone Note model
- Minor naming fixes in methods and classNames

Build instructions:
- Delete atom_shell & src/node_modules
- cd src && apm install .
- grunt build --scratch

### CODE REVIEW

File: src/renderer/stores/note.js [hunk 1]
```diff
@@ -15,6 +13,44 @@ var Note = Backbone.Model.extend({
...
+  },
+
+  updateBlock: function(payload) {
+    var i = payload.content.index;
+    var blocks = this.get('blocks');
+    blocks[i] = {
+      language: payload.content.language,
+      content: payload.content.content || ''
+    };
+    this.set('blocks', blocks);
```
Comment (ghost) [inline src/renderer/stores/note.js:51]: What happens if you close the editor without saving? Will it keep the updated block changes?

File: src/renderer/components/content/content.js [hunk 2]
```diff
@@ -15,15 +15,15 @@ var Content = React.createClass({
     if(this.props.view === 'Notes') {
```
Comment (Cfeusier) [inline src/renderer/components/content/content.js]: style guide violation :)

File: src/renderer/stores/note.js (hunk 1 above)
Comment (d4ncer) [inline src/renderer/stores/note.js:51]: No, currentNote loses reference to that Backbone model, so it gets GC'd.

Comment (minocys) [inline src/renderer/stores/note.js:51]: The editor was never designed to have persistent memory in and off itself. All local changes to the backbone models/collections will disappear if the program is closed without a save.

**Pull request merged**